test_runner/batch_others/test_backpressure.py

```diff
@@ -0,0 +1,159 @@
+from contextlib import closing, contextmanager
+import psycopg2.extras
+from fixtures.zenith_fixtures import ZenithEnvBuilder
+from fixtures.log_helper import log
+import os
+import time
+import asyncpg
+from fixtures.zenith_fixtures import Postgres
+import threading
+
+pytest_plugins = ("fixtures.zenith_fixtures")
+
+
+@contextmanager
+def pg_cur(pg):
+    with closing(pg.connect()) as conn:
+        with conn.cursor() as cur:
+            yield cur
+
+
+# Periodically check that all backpressure lags are below the configured threshold,
+# assert if they are not.
+# If the check query fails, stop the thread. Main thread should notice that and stop the test.
+def check_backpressure(pg: Postgres, stop_event: threading.Event, polling_interval=5):
+    log.info("checks started")
+
+    with pg_cur(pg) as cur:
+        cur.execute("CREATE EXTENSION zenith")  # TODO move it to zenith_fixtures?
+
+        cur.execute(
+            "select setting::bigint*1024*1024 from pg_settings where name='max_replication_write_lag'"
+        )
+        res = cur.fetchone()
+        max_replication_write_lag_bytes = res[0]
+        log.info(f"max_replication_write_lag: {max_replication_write_lag_bytes}")
+
+        cur.execute(
+            "select setting::bigint*1024*1024 from pg_settings where name='max_replication_flush_lag'"
```

Comment:
```suggestion
            "select pg_size_bytes(current_setting('max_replication_flush_lag'))"
```
I still feel burned by our bytes vs GB mixup with the cluster size limit. This works without knowing the setting's base unit.

Reply:
Thanks. Fixed